server/src/ai/ai.functions.ts

```diff
@@ -22,27 +22,104 @@ const openai = new OpenAI({
   apiKey: process.env.OPENAI_API_KEY,
 });
 
-/** makeSummary. 
+/** summarize. 
  * 
  * Takes in all reviews from a course as text and 
- * creates a 50 word summary of those reviews.
+ * generates 5 tags for those reviews describing lectures, assignments, professor,
+ * skill, and resources as well as the corresponding connotation of that tag.
  * @params a string that combines all reviews from a course
- * @returns summary of reviews
+ * @returns a dictionary containing 
+ * Summary: 50 word summary of all reviews,
+ * Tags: array of nouns, adjectives, and their corresponding connotations describing
+ * in that order, ex: ["Lectures", "Entertaining", "Positive"]
  */
-async function makeSummary(text: string) {
+async function summarize(text: string) {
   const completion = await openai.chat.completions.create({
     model: "gpt-3.5-turbo",
     messages: [
-      { role: "system", content: "You are creating a 50 word summary based on the collection of course reviews provided." },
+      {
+        role: "system", content: `
+          You are given a collection of course reviews provided where each review is separated by a /. You
+          will then complete two tasks. First you should generate a 50 word summary of all reviews. Then 
+          you should create 5 adjectives describing the lectures, assignments, professor, skills, and resources,
+          along with their connotations (positive, negative, neutral). Please only pick one adjective for each.
+          
+          Please provide the summary and tags in the following format:
+          Summary: [50-word summary]
+          Tags:
+          'Lectures: [adjective] (positive/negative/neutral),
+          Assignments: [adjective] (positive/negative/neutral),
+          Professor: [adjective] (positive/negative/neutral),
+          Skills: [adjective] (positive/negative/neutral),
+          Resources: [adjective] (positive/negative/neutral)'.
+        `
+      },
       { role: "user", content: text }
     ],
   });
-  return completion.choices[0].message.content;
+
+  const response = completion.choices[0].message.content;
+  const summaryMatch = response.match(/Summary: ([\s\S]*?)(?=Tags)/);
+  const summary = summaryMatch ? summaryMatch[1].trim() : "Summary not found.";
+  const tagsMatch = response.match(/Tags:\s*([\s\S]*)/);
+  const tags = tagsMatch ? tagsMatch[1] : "";
+  const tagsObject: { [key: string]: [string, string] } = {};
+  tags.split(',').forEach(item => {
+    const match = item.match(/(\w+): (.+) \((.+)\)/);
+    if (match) {
+      const category = match[1].trim();
+      const adjective = match[2].trim();
+      const connotation = match[3].trim();
+      if (adjective !== "N/A" && connotation != "N/A") {
+        tagsObject[category] = [adjective, connotation];
```

Comment:
Also here, how does the error handling work in the future if there are less than 5 tags or if we have trouble parsing the tags for a certain class?

Reply:
I was thinking to handle this in the front end, so it'll only display a tag if there was something generated for that?